File: Core/Models/AuditLog.swift
import Foundation

public nonisolated struct AuditEntry: Codable, Identifiable, Hashable, Sendable {
    public let id: EntityID
    public var at: Date
    public var actorUserID: EntityID
    public var action: String
    public var targetEntity: String
    public var targetID: EntityID
    public var changes: [String: String]

    public init(
        id: EntityID = UUID(),
        at: Date = Date(),
        actorUserID: EntityID,
        action: String,
        targetEntity: String,
        targetID: EntityID,
        changes: [String: String] = [:]
    ) {
        self.id = id
        self.at = at
        self.actorUserID = actorUserID
        self.action = action
        self.targetEntity = targetEntity
        self.targetID = targetID
        self.changes = changes
    }
}

File: flutter_app/lib/core/models/audit_log.dart
// Port of Core/Models/AuditLog.swift.
// Pure data — no Flutter imports (logic-layer rule).

import 'entity_id.dart';

class AuditEntry {
  final EntityID id;
  final DateTime at;
  final EntityID actorUserId;
  final String action;
  final String targetEntity;
  final EntityID targetId;
  final Map<String, String> changes;

  const AuditEntry({
    required this.id,
    required this.at,
    required this.actorUserId,
    required this.action,
    required this.targetEntity,
    required this.targetId,
    this.changes = const {},
  });

  factory AuditEntry.fromJson(Map<String, dynamic> json) => AuditEntry(
    id: json['id'] as String,
    at: DateTime.parse(json['at'] as String),
    actorUserId: json['actorUserID'] as String,
    action: json['action'] as String,
    targetEntity: json['targetEntity'] as String,
    targetId: json['targetID'] as String,
    changes:
        (json['changes'] as Map<String, dynamic>?)?.map(
          (k, v) => MapEntry(k, v as String),
        ) ??
        {},
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'at': at.toIso8601String(),
    'actorUserID': actorUserId,
    'action': action,
    'targetEntity': targetEntity,
    'targetID': targetId,
    'changes': changes,
  };
}
